llm/gpt-2/README.md

```diff
@@ -0,0 +1,58 @@
+# Run GPT-2 in llm.c on any cloud with SkyPilot
+
+This is a reproducible package of llm.c's GPT-2 (124M) training by @karpathy (https://github.com/karpathy/llm.c/discussions/481)
+With SkyPilot, you can run GPT-2 (124M) training on any cloud.
+
+## Prerequisites
+
+1. Install [SkyPilot](https://github.com/skypilot-org/skypilot):
+```bash
+pip install skypilot-nightly
+```
+2. Enable clouds for SkyPilot:
+```bash
+sky check
+```
+Please check the instructions for enabling clouds at [SkyPilot doc](https://skypilot.readthedocs.io/en/latest/getting-started/installation.html).
+3. Download the YAMLs in this directory for data processing and training:
+```bash
+wget https://github.com/skypilot-org/skypilot/blob/master/llm/gpt-2/gpt2-data.yaml
+wget https://github.com/skypilot-org/skypilot/blob/master/llm/gpt-2/gpt2-train.yaml
+```
+
+## Data processing
+
+Run the following command to process the training data on a CPU VM and store it in a cloud bucket for future use (replace `your-bucket-name` with your bucket name):
+```bash
+sky launch -c gpt2-data gpt2-data.yaml --env BUCKET_NAME=your-bucket-name
+```
+
+
+## Training
+
+After the data is processed, you can then train the model on a GPU VM with 8 A100 GPUs (replace `your-bucket-name` with your bucket name):
+
+```bash
+sky launch -c gpt2-train --detach-setup gpt2-train.yaml --env BUCKET_NAME=your-bucket-name
+```
+
+Or, you can train the model with a single A100, by adding `--gpu A100`:
+```bash
+sky launch -c gpt2-train --detach-setup gpt2-train.yaml --gpu A100 --env BUCKET_NAME=your-bucket-name
+```
+
```

Comment:
nit: if we have graphs similar to karpathy's, it will be nice to put them here :) No worries if not. 

![image](https://github.com/skypilot-org/skypilot/assets/4416605/bb14cce2-6dc1-4a7b-9d6d-628acb1b025b)

Reply:
Added the loss for training. The eval figure requires additional dependency which I unfortunately did not installed before the current training.

![](https://imgur.com/lskPEAQ.png)